internal/pkg/config/matcher_test.go

```diff
@@ -0,0 +1,244 @@
+package config
```

Comment:
I think you've got good test cases in general, but the tests are also fairly high level, and as such I'm not confident that every case is handled. It also feels like there's a good amount of overlap between cases: all of these tests end up exercising `matchEither` indirectly, and that's where the tricky logic is. I think it would be good to have tests for `matchEither` specifically.

Reply:
Low level unit tests added for the various matcher types.